src/resolvers-cassandra/Settings/mutations.js

```diff
@@ -3,21 +3,106 @@
+      .catch(reject);
+  });
+}
+
+const insertTopics = trackEvent(_insertTopics, 'Settings.Topics.Insert', (response, err) => ({numTopicsInserted: err ? 0 : response.numTopicsInserted}));
+
 /**
- * @param {{input: {targetBbox: number[], defaultZoomLevel: number, logo: string, title: string, name: string, defaultLocation: number[], storageConnectionString: string, featuresConnectionString: string, mapzenApiKey: string, fbToken: string, supportedLanguages: string[]}}} args
- * @returns {Promise.<{name: string, properties: {targetBBox: number[], defaultZoomLevel: number, logo: string, title: string, defaultLocation: number[], storageConnectionString: string, featuresConnectionString: string, mapzenApiKey: string, fbToken: string, supportedLanguages: string[]}}>}
+ * @param {{input: {siteType: string, targetBbox: number[], defaultZoomLevel: number, logo: string, title: string, name: string, defaultLocation: number[], storageConnectionString: string, featuresConnectionString: string, mapzenApiKey: string, fbToken: string, supportedLanguages: string[]}}} args
+ * @returns {Promise}
  */
-function createOrReplaceSite(args, res) { // eslint-disable-line no-unused-vars
+function createSite(args, res) { // eslint-disable-line no-unused-vars
+  return new Promise((resolve, reject) => {
+    const siteType = args && args.input && args.input.siteType;
+    if(!siteType || !siteType.length) {
+      return reject(`siteType for sitename ${args.name} is not defined`);
+    }
+    cassandraConnector.executeQuery('SELECT * FROM fortis.sitesettings WHERE sitename = ?', [args.name])
+      .then(rows => {
+        if(!rows || !rows.length) {
+          return insertTopics(siteType);          
+        } else {
+          return reject(`Site with sitename ${args.name} already exists.`);
+        }
+      })
+      .then(() => {
+        return cassandraConnector.executeBatchMutations([{
+          query: `INSERT INTO fortis.sitesettings (
+            sitetype, 
+            sitename,
+            geofence,
+            languages,
+            defaultzoom,
+            title,
+            logo,
+            insertion_time
+          ) VALUES (?,?,?,?,?,?,?,?,dateof(now()))`,
+          params: [
+            args.siteType,
+            args.name,
+            args.targetBbox,
+            args.supportedLanguages,
+            args.defaultZoomLevel,
+            args.title,
+            args.logo
+          ]
+        }]);
+      })
+      .catch(reject);
+  });
+}
+
+/**
+ * @param {{input: {siteType: string, targetBbox: number[], defaultZoomLevel: number, logo: string, title: string, name: string, defaultLocation: number[], storageConnectionString: string, featuresConnectionString: string, mapzenApiKey: string, fbToken: string, supportedLanguages: string[]}}} args
+ * @returns {Promise}
+ */
+function removeSite(args, res) { // eslint-disable-line no-unused-vars
+  return new Promise((resolve, reject) => {
+    cassandraConnector.executeBatchMutations([{
```

Comment:
I wonder if it's possible to just return the `cassandraConnector.executeBatchMutations` here instead of wrapping it in a promise? The cassandraConnector already returns a promise and we're not doing any post-processing, so I think this code should be equivalent :)

```
function removeSite(args, res) {
  return cassandraConnector.executeBatchMutations([{...}]);
}
```